src/Agent/NewRelic/Agent/Core/NewRelic.Agent.Core/AgentManager.cs

```diff
@@ -69,15 +69,8 @@ public static IAgentManager Instance
         {
             get
             {
-                try
-                {
-                    return Singleton.ExistingInstance;
-                }
-                catch (NullReferenceException)
-                {
-                    // The singleton pointer may be null if we instrument a method that's invoked in the agent constructor.
-                    return DisabledAgentManager;
-                }
+                // The singleton pointer may be null if we instrument a method that's invoked in the agent constructor.
+                return Singleton?.ExistingInstance ?? DisabledAgentManager;
```

Comment:
Not related to this task but I got the ok during pairing last week.

Reply:
It might be the case that the previous code relied on the try catch statement instead of the conditional to avoid needing to evaluate the conditional every time this code is executed. It might have been a performance optimization because the problem only occurs during the shorter period where the agent constructor is still executing.

Reply:
I suppose that could be, though I'd be surprised if it was a measurable difference. I can wrap it in a `DEBUG` if you think that would be better.